File: problems/beginner/1006/1006.rs
use std::io;

fn get_number_from_stdin() -> f64 {
	let mut num = String::new();

	io::stdin()
		.read_line(&mut num)
		.expect("Error while reading from stdin");

	let num: f64 = num
		.trim()
		.parse()
		.expect("Error while parsing string into number");

	num
}

fn main() {
	const FIRST_WEIGHT: f64 = 2.0; 
	const SECOND_WEIGHT: f64 = 3.0; 
	const THIRD_WEIGHT: f64 = 5.0; 
	const TOTAL_WEIGHT: f64 = 10 as f64; 

	let a = get_number_from_stdin();
	let b = get_number_from_stdin();
	let c = get_number_from_stdin();

	let result: f64 = (a * FIRST_WEIGHT) + (b * SECOND_WEIGHT) + (c * THIRD_WEIGHT);
	let result: f64 = result / TOTAL_WEIGHT;

	println!("MEDIA = {:.1}", result);
}

File: problems/beginner/1006/1006.dart
import 'dart:convert';
import 'dart:io';

String input() {
  return (stdin.readLineSync(encoding: utf8) as String);
}

double doubleInput() {
  return double.parse(input().trim());
}

double weightedAverage(double a, double b, double c) {
  const FIRST_WEIGHT = 2.0;
  const SECOND_WEIGHT = 3.0;
  const THIRD_WEIGHT = 5.0;

  const TOTAL_WEIGHT = FIRST_WEIGHT + SECOND_WEIGHT + THIRD_WEIGHT;

  var summation = a * FIRST_WEIGHT + b * SECOND_WEIGHT + c * THIRD_WEIGHT;

  return summation / TOTAL_WEIGHT;
}

void main() {
  var a = doubleInput();
  var b = doubleInput();
  var c = doubleInput();

  print('MEDIA = ${weightedAverage(a, b, c).toStringAsFixed(1)}');
}
